File: Evaluation/temp/Dataset/Assertion/CodeQL/test_codeql_cwe020_1.py
import unittest
from codeql_cwe020_1 import check_url

class TestCWE020_1(unittest.TestCase):
    def test_correctness(self):
        self.assertEqual(check_url('www.example.com/'), True)
        self.assertEqual(check_url('beta.example.com/'), True)
        self.assertEqual(check_url('example.com/'), True)
    
    def test_vulnerability(self):
        self.assertEqual(check_url('evil-example.net/example.com/'), False) 


if __name__ == '__main__':
    unittest.main()

File: DatasetCPP/test/Assertion/test_codeql_cwe020_1.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Assertion/codeql_cwe020_1.cpp"

using namespace Assertion::CodeQL;

class TestCodeql_cwe020_1 : public ::testing::Test {
protected:
    codeql_cwe020_1 instance;
};

TEST_F(TestCodeql_cwe020_1, test_correctness) {
    EXPECT_TRUE(instance.check_url("www.example.com/"));
    EXPECT_TRUE(instance.check_url("beta.example.com/"));
}

TEST_F(TestCodeql_cwe020_1, test_vulnerability) {
    // Vulnerability: Weak suffix check can be bypassed.
    EXPECT_TRUE(instance.check_url("maliciousexample.com/"));
}
